### ISSUE TITLE
Accept concrete (unused) implementations of public abstract methods

### ISSUE THREAD

**Issue by Jym77** [webpro/knip#360]
I think knip should accept the concrete implementation of an abstract method, even when the concrete version is not used directly, when the abstract method is either `@public`, or used.

foo.ts:
```typescript
export abstract class AFoo {
  /** @public */
  public abstract foo(): number
}

export class Foo {
  public foo() {
    return 0
  }
}
```

index.ts:
```typescript
import { AFoo, Foo } from "./foo";

const x = new Foo()

function f(x: AFoo): number {
  return x.foo()
}
```

When knipping that, I got:
```
Unused exported class members (1)
foo  Foo  src/foo.ts
```

But `Foo#foo` is pretty much needed, and even if function `f` wasn't here, it would be required to implement the abstract method which I've already told knip is meant for external API.
Adding a release tag on `Foo#foo` is not really convenient since I have around 40 of these subclasses in my actual case 🙈

**Comment by webpro** [comment]
How about this:

```ts
export abstract class AFoo {
  public abstract foo(): number
}

export class Foo implements AFoo {
  public foo() {
    return 0
  }
}
```

```ts
import { Foo } from "./foo";

const x = new Foo()

function f(x: Foo): number {
  return x.foo()
}
```

**Comment by Jym77** [comment]
🙈 
Yep, forgot that in the MNWE…

**Comment by Jym77** [comment]
🤔 I'm still getting it in my large real code, but can't make a MNWE about it. That might be another root cause that my initial analysis…

**Comment by Jym77** [comment]
Managed to get a MNWE: https://codesandbox.io/p/devbox/mystifying-swanson-ytr3wq

The classes `After` (in its own file) ad `Before` (in the same file as the parent abstract class) are exactly the same, but `After` is (incorrectly, as it needs to implement the public abstract from `Selector`) reported for unused `[Symbol.iterator]` while `Before` isn't.

**Comment by webpro** [comment]
I still don't see actual usage of the member reported as unused? I see classes, no instances.

**Comment by webpro** [comment]
For now:

- Unused exports in entry files are not reported (you can enable this with `--include-entry-exports`)
- Members of exports in entry files are also not reported. However, `--include-entry-exports` does not enable this for members, so I think this is a bug I'm going to fix. You saw that right, thanks!
- `Before` is not exported, so it's not reported. This is ESLint (or other isolated file linter) area :)

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.3.1. See [Release 5.3.1](https://github.com/webpro/knip/releases/tag/5.3.1) for release notes._
